packages/@aws-cdk/aws-ecs-patterns/lib/base/load-balanced-service-base.ts

```diff
@@ -108,11 +120,21 @@ export abstract class LoadBalancedServiceBase extends cdk.Construct {
 
   public readonly targetGroup: elbv2.ApplicationTargetGroup | elbv2.NetworkTargetGroup;
 
+  public readonly cluster: ecs.ICluster;
+
   public readonly logDriver?: ecs.LogDriver;
 
   constructor(scope: cdk.Construct, id: string, props: LoadBalancedServiceBaseProps) {
     super(scope, id);
 
+    if (props.cluster && props.vpc) {
+      throw new Error(`You can only specify either vpc or cluster. Alternatively, you can leave both blank`);
+    } else if (props.cluster) {
+      this.cluster = props.cluster;
+    } else {
+      this.cluster = new ecs.Cluster(this, 'Cluster', { vpc: props.vpc });
```

Comment:
+1 that this makes much more sense for Fargate then for EC2, so I rather optimize for this use case.

As for default cluster, I am not aware that we have a concept like that, but should be fairly easy to introduce:

```ts
function getCreateDefaultCluster(scope: Construct, vpc?: ec2.Vpc): ecs.Cluster {
  const DEFAULT_CLUSTER_ID = 'EcsDefaultCluster<plug-in-some-uuid>';
  const stack = Stack.of(this);
  return stack.tryGetChild(id) as ecs.Cluster || new ecs.Cluster(stack, id, { vpc });
}
```

Reply:
As the properties are set and only available in the base I added this code here. I am open to moving this to the FargateService.

As I understood it, the concept of defaults is part of another issue or PR

Reply:
ECS has a concept of a cluster literally called 'default'.  The console first-run will create it, as will any ECS-optimized AMI-based instances that don't have another cluster name configured.  So, many but not all accounts have a 'default' cluster already.

Could we have something like the ImportedVpc and Vpc.fromLookup concepts for ECS clusters?  i.e. a way to check if a cluster named 'default' exists in the account/region, and if not, add it to the stack.

Reply:
> +1 that this makes much more sense for Fargate then for EC2, so I rather optimize for this use case.
> 
> As for default cluster, I am not aware that we have a concept like that, but should be fairly easy to introduce:
> 
> ```ts
> function getCreateDefaultCluster(scope: Construct, vpc?: ec2.Vpc): ecs.Cluster {
>   const DEFAULT_CLUSTER_ID = 'EcsDefaultCluster<plug-in-some-uuid>';
>   const stack = Stack.of(this);
>   return stack.tryGetChild(id) as ecs.Cluster || new ecs.Cluster(stack, id, { vpc });
> }
> ```

https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_CreateCluster.html - 

clusterName - The name of your cluster. If you do not specify a name for your cluster, you create a cluster named default. Up to 255 letters (uppercase and lowercase), numbers, and hyphens are allowed.

Reply:
I think this sounds cool, but I like that I can generate the CFN from CDK without knowledge of the account and then deploy it. This lookup would then no longer be possible, would it?

Reply:
@hoegertn my concern with creating a new cluster per FargateService by default is the resulting experience in the console and CLI.  Outside of maybe a demo, it doesn't seem like a desirable state to have 1 cluster per service.  In the console, you will have to select among your many clusters, and then click again into the one service in that cluster.  Based on most customers' behavior, the desirable state is for all Fargate services to belong to one cluster.

Reply:
I fully agree with this. If there is a great way to have this default cluster I am totally fine with it. But I like if CDK is not calling AWS before creating the template.

Reply:
Since the default cluster has a static name 'default', another option could be to always use the 'default' cluster for the Fargate service if the cluster is not provided.  Con: this would require the user to create the 'default' cluster manually in the account/region (through console first-run, etc), otherwise the stack creation will fail due to non-existent cluster.

Reply:
FWIW, the VPC construct is already describing stuff about the account, like availability zones

Reply:
We could implement a context provider for ECS clusters, probably worth to do generally. However, for this use case, since the default cluster must be created manually, I'd argue that we can simply define a CDK default cluster *per stack* in the method I described, as it will seamlessly work for any user, and will not create any conflicts between stacks/apps/users within the same account/region.

Reply:
Ok, I will implement the default cluster per stack approach. I think one cluster per stack is a reasonable trade-off.